### PR TITLE
Regard continental bonuses when evaluating initial placement

### PR DESCRIPTION
**Pull request by AntonOellerer**

### CODE REVIEW

File: src/main/java/agents/leeroy/GameUtils.java [hunk 1]
```diff
@@ -118,13 +125,48 @@ private static Integer evaluateInitialBoard(Node node, Risk risk) {
         return continents;
     }
 
+    private static int getTotalContinentBonus(int playerId, RiskBoard riskBoard) {
```
Comment (freecraver) [inline src/main/java/agents/leeroy/GameUtils.java:128]: interesting method :-)

**Review by freecraver — APPROVED** [review]
looks good. small thing I noticed: if you increase the timeout by a lot the result still stays the same. This might be an indicator that the mcts (in its current form) is overkill as it leads to the same solution as a greedy algorithm. But I am unsure about this 🙃

File: src/main/java/agents/leeroy/GameUtils.java (hunk 1 above)
Comment (AntonOellerer) [reply src/main/java/agents/leeroy/GameUtils.java:128]: Yeah, I wanted to have a uniform interface for bonus and malus, so I created this intermediate method :sweat_smile:

**Comment by AntonOellerer** [comment]
What do you mean by "the results"?

**Comment by freecraver** [comment]
Well, if you increase `INITIAL_SELECT_TIMEOUT_PENALTY` in `Leeroy.java` to e.g. 1000 the chosen actions stay the same, i.e. the same territories are selected

**Comment by AntonOellerer** [comment]
But if you do this, doesn't the Algorithm still have the same processing time?

**Comment by freecraver** [comment]
Nope it affects `shouldStopComputation` and therefor stops earlier, i.e. reduces exploration

**Comment by AntonOellerer** [comment]
Might be that a good option is at the beginning of the search tree, which would indeed lead to this behaviour

**Comment by freecraver** [comment]
Yes I'm just a bit sceptical as in my simulations the initial placement doesn't seem to take the opponent's moves into account a lot (but maybe that's a better strategy, idk) 🤷

**Comment by AntonOellerer** [comment]
Yeah, it just tries to avoid the opponent getting too much initial bonuses.
You might observe such static behaviour since the algorithm tries to reduce the amount of areas and foreign borders, so once it settles on a "strategy" it sticks to it.
All in all, it is not a very dynamic algorithm, but we could talk about overhauling it ofc.

**Comment by freecraver** [comment]
Ok - guess there are other parts of the agent where we can improve more. So please go ahead & merge :-)

**Pull request merged**